### ISSUE TITLE
Upgrade Centos to Centos 9

### ISSUE THREAD

**Issue by pli888** [gigascience/gigadb-website#1897]
## User stories

>As s developer,
>I want to run the latest version of Centos on our infrastructure
>so that we reduce our exposure to legacy vulnerabilities
>and so that we don't run into compatibility issues

# Acceptance criteria

>Given infrastructure configuration has been updated
>When we deploy a new infrastructure
>Then it is running Centos 9 

## Additional information

Centos 8 Stream end of life is 31st May 2024

We are using Centos 8 stream release 8 for both bastion and application (aws-instance.tf) servers but Centos Stream 8 is coming to its end of life at the end of this month. [Information](https://lists.centos.org/hyperkitty/list/devel@lists.centos.org/thread/BIWCCSDWOPCX5DMDEYJVTTEXHNYHSJZ3/) about this event says there will be no updates including those relating to security.

The [blog post](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) reporting this news suggests moving to CentOS Stream 9 as an option.

**Comment by pli888** [comment]
I got this error message yesterday when I executed the bastion playbook to provision a new bastion EC2 server:
```
TASK [../../roles/centos-eol-fix : Install epel-release] **********************************************************************
fatal: [10.99.0.5]: FAILED! => {"changed": false, "msg": "Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist", "rc": 1, "results": []}
```

I managed to resolve this error by updating `ops/infrastructure/roles/centos-eol-fix/tasks/main.yml`:
```
- name: Replace release version from Centos 8 to 8-Stream
  copy:
    content: "8-stream"
    dest: /etc/yum/vars/releasever
    owner: root
    group: root
    mode: 0644

- name: Comment out mirrorlist because Centos 8 Stream is EOL
  shell: sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*

- name: Use vault URL because Centos 8 Stream is EOL
  shell: sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*

- name: Delete cache
  shell: dnf clean all && rm -r /var/cache/dnf
```

**Comment by pli888** [comment]
Changes will be needed if we decide to upgrade to [Centos Stream 9 AMI](https://www.centos.org/download/aws-images/). The user will change to  `ec2-user` instead of `centos` which will mean updating provisioning scripts. Also, there are [reports](https://serverfault.com/questions/1142858/hostbased-authentication-fail-on-centos-stream-9) saying that using rsa keys to SSH into Centos 9 Stream servers is not possible so will need to use ed25519 based keys.

**Comment by rija** [comment]
@pli888, @kencho51, @alli83 
This is significant infrastructure change that requires tearing down and replace our current EC2 instances on live infrastructure, so we have to wait until the blue/green deployment is ready (#1803) before we take on this ticket.

**Comment by kencho51** [comment]
Close as implemented in PR #1998
